### ISSUE TITLE
Simplify AgentOption

### ISSUE THREAD

**Issue by aoberoi** [slackapi/node-slack-sdk#735]
### Description

The `AgentOption` is currently defined as follows:

```typescript
export type AgentOption = Agent | {
  http?: Agent,
  https?: Agent,
} | boolean;
```

This allows for a lot of flexibility, but much of that flexibility isn't possible to use. So it ends up being needless complexity.

First, using a `boolean` is only meaningful if you want to specify `false` as the value. In node.js, this would mean that the globalAgent is ignored and a new agent is created. In practice, there's no use case for that behavior to be desired.

Next, the `{ http?: Agent; https?: Agent }` interface allows for separate agents for when the specific URL is in one or the other scheme. However, there's only ever one `slackApiUrl`, so there's no need for this kind of switching. The only exception would be for the WebSocket connection in `RTMClient`. In that case, the request would always be over `https` (since Slack will always respond with this URL), so there's still no point in offering an interface that allows switching.

In conclusion, we can simplify the `AgentOption` to just `Agent`.

### Requirements (place an `x` in each of the `[ ]`)
* [x] I've read and understood the [Contributing guidelines](https://github.com/slackapi/node-slack-sdk/blob/master/.github/contributing.md) and have done my best effort to follow them.
* [x] I've read and agree to the [Code of Conduct](https://slackhq.github.io/code-of-conduct).
* [x] I've searched for any related issues and avoided creating a duplicate issue.
